### PR TITLE
fix(judicial-system): Fix issues with the police case files list

### PR DESCRIPTION
**Pull request by oddsson**
# Fix issues with the police case files list

https://app.asana.com/0/1199153462262248/1201312869488037/f

## What

Display a message in the police case files list if

- the case was not found in LÖKE
- the case was found in LÖKE but has no files

## Why

Better UX

### When a case is not found in LÖKE
Screen Shot 2021-11-09 at 14 44 23

### When  there is a server error
Screen Shot 2021-11-09 at 14 43 42

### Otherwise it looks like this
https://user-images.githubusercontent.com/3789875/140946474-9b0097a4-6caf-4f01-9cf3-1e090c44c277.mov

NOTE: I couldn't test when a case is found in LÖKE but has no files because there is currently no such LÖKE case.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFiles.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ export interface PoliceCaseFilesData {
   files: PoliceCaseFile[]
   isLoading: boolean
   hasError: boolean
+  errorMessage?: string
```
Comment (telmag92) [inline apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFiles.tsx:22]: This errorMessage is unused, isn't it missing in setPoliceCaseFiles ?

File: apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFilesForm.tsx [hunk 2]
```diff
@@ -178,24 +179,86 @@ const CaseFilesForm: React.FC<Props> = (props) => {
...
-                    <Checkbox
-                      name="selectAllPoliceCaseFiles"
-                      label={formatMessage(
-                        m.sections.policeCaseFiles.selectAllLabel,
+                    <LoadingDots />
+                  </Box>
+                ) : policeCaseFiles?.hasError ? (
+                  policeCaseFiles?.errorMessage &&
+                  policeCaseFiles?.errorMessage.indexOf('404') > -1 ? (
+                    <Box
```
Comment (telmag92) [inline apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFilesForm.tsx:212]: This box is duplicated 3 times depending on the error, wondering if we could make it more reusable.

Also these errors give the exact same warning box:
policeCaseFiles?.files.length === 0  and 
policeCaseFiles?.errorMessage && policeCaseFiles?.errorMessage.indexOf('404') > -1

File: apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFilesForm.tsx [hunk 3]
```diff
@@ -235,32 +298,25 @@ const CaseFilesForm: React.FC<Props> = (props) => {
...
-                  <Text variant="h5">
-                    {formatMessage(m.sections.policeCaseFiles.errorMessage)}
-                  </Text>
-                </Box>
-              ) : (
-                <Box display="flex" alignItems="center" paddingY={3}>
-                  <Box display="flex" marginRight={2}>
-                    <Icon icon="checkmark" color="blue400" />
+                ) : (
+                  <Box
```
Comment (telmag92) [inline apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFilesForm.tsx:302]: Same goes for this box, only difference is the icon values and message

**Review by telmag92 — COMMENTED** [review]
Few thoughts regarding duplication on the box with icon and text

File: apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFiles.tsx (hunk 1 above)
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFiles.tsx:22]: I believe we're using the errorMessage in line 209 in caseFilesForm

```
policeCaseFiles?.hasError ? (
    policeCaseFiles?.errorMessage &&
    policeCaseFiles?.errorMessage.indexOf('404') > -1 ? (
...
```

Am I missing something?

Comment (oddsson) [reply apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFiles.tsx:22]: This was resolved irl

**Review by telmag92 — APPROVED** [review]

**Pull request merged**